lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -100,12 +101,12 @@ def __init__(self, config: PbenchServerConfig, logger: Logger, *schemas: ApiSche
             raise MissingDatasetNameParameter(
                 api_name, "dataset parameter is not defined or not required"
             )
-        if self.schema.authorization != API_AUTHORIZATION.DATASET:
+        if self.schema.authorization != ApiAuthorizationType.DATASET:
             raise MissingDatasetNameParameter(
                 api_name, "schema authorization is not by dataset"
             )
 
-    def preprocess(self, params: ApiParams) -> CONTEXT:
+    def preprocess(self, params: ApiParams, context: ApiContext) -> ApiContext:
```

Comment:
This function, contrary to the assertion of the type hint, does not return a value.

(And, you might want to update the reference to "`CONTEXT`"` at line 112.